src/components/Layout/RightPaneLayout/RightPaneLayout.stories.tsx

```diff
@@ -0,0 +1,158 @@
+import { Meta } from "@storybook/react";
+import { PropsWithChildren, useMemo } from "react";
+import { zeroTo } from "src/utils/sb";
+import { Css, Palette } from "../../../Css";
+import { Button } from "../../Button";
+import { IconButton } from "../../IconButton";
+import { GridColumn, GridDataRow, GridTable, simpleHeader, SimpleHeaderAndData } from "../../Table";
+import { FullBleed } from "../FullBleed";
+import { PreventBrowserScroll } from "../PreventBrowserScroll";
+import { ScrollableContent } from "../ScrollableContent";
+import { ScrollableParent } from "../ScrollableParent";
+import { RightPaneProvider } from "./RightPaneContext";
+import { RightPaneLayout } from "./RightPaneLayout";
+import { useRightPane } from "./useRightPane";
+
+export default {
+  component: RightPaneLayout,
+  title: "Workspace/Components/Layout/RightPaneLayout",
+} as Meta;
+
+function SampleContent() {
+  const { openRightPane } = useRightPane();
+  return (
+    <div css={Css.bgWhite.h100.$}>
+      <Button label={"Open Pane"} onClick={() => openRightPane({ content: <DetailPane /> })} />
+    </div>
+  );
+}
+
+function DetailPane() {
+  const { closeRightPane } = useRightPane();
+  return (
+    <div css={Css.bgWhite.h100.$}>
+      <Button label={"Close Pane"} onClick={() => closeRightPane()} />
+    </div>
+  );
+}
+
+export function Basic() {
+  return (
+    <RightPaneProvider>
+      <div css={Css.df.fdc.gap2.h100.$}>
+        <RightPaneLayout paneBgColor={Palette.White}>
+          <SampleContent />
+        </RightPaneLayout>
+      </div>
+    </RightPaneProvider>
+  );
+}
+
+export function GridTableWithRightPane() {
+  return (
+    <TestProjectLayout>
+      <ExamplePageComponent />
+    </TestProjectLayout>
+  );
+}
+
+function ExamplePageComponent() {
+  return (
+    <>
+      <FullBleed>
+        <header css={{ ...Css.py2.bb.bGray200.$ }}>
+          <h1 css={Css.xlSb.$}>Page content fixed to top</h1>
+        </header>
+      </FullBleed>
+      <ScrollableTableExample />
+    </>
+  );
+}
+
+function TestProjectLayout({ children }: PropsWithChildren<{}>) {
+  return (
+    <PreventBrowserScroll>
+      <div css={Css.df.h100.$}>
+        <ScrollableParent xss={Css.px3.$}>{children}</ScrollableParent>
+      </div>
+    </PreventBrowserScroll>
+  );
+}
+
+function ScrollableTableExample({ numCols, numRows }: { numCols?: number; numRows?: number }) {
+  return (
+    <ScrollableContent>
+      <RightPaneProvider>
```

Comment:
Currently exposing `RightPaneProvider` as in testing on schedules it needs to sit at a higher level than the layout part of the component. Looking for options here as to this being a pattern we are okay with to keep the context provider localized to the pages that need it. Currently the Schedules detail pane is shared between the different views (Table, Gantt, Calendar) and the layout needs to support being passed by the `ScrollingContainer`.

Reply:
Fwiw I agree ideally a caller could just use `RightPaneLayout` and not have to worry about "I also need to wrap that in `RightPaneProvider`".

I had the same musing, of having `RightPaneLayout` just internally include a new `RightPaneProvider`, but if it needs to be higher up in the tree, maybe we just add `RightPaneProvider` directly to `BeamContext`? 

Granted, we have a ton of `SnackbarProvider` / `ToastProvider` / `AutoSaveStatusProvider` already there :-) but :shrug: so far its been working out well.

Reply:
I agree, the initial thought was to have them both contained in the singular beam component but the complexity of consuming it in blueprint has proven to be more tricky. I am testing to see how things work with exposing it via the BeamContext which for now seems like the path of least resistance.